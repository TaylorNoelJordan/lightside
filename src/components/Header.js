import React from 'react';
import { Link } from 'react-router-dom';
import './Header.scss';

const Header = (props) => {
    const { favorites } = props;


    return (
        <header>
            <Link to='/'>
                <h1>LightSide</h1>
            </Link>
            <Link to='/Favorites'>
                <button className='faveBtn'>View Favorites: {favorites}</button>
            </Link>
        </header>
    )
}

export default Header;