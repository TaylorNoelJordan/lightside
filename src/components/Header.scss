header {
    padding: 30px;
}

h1 {
    color: #feda4a;
    font-size: 40px;
    font-family: 'Galaxy 1', sans-serif;
    margin-top: 0;
    text-transform: uppercase;
}

.faveBtn {
    border-radius: 15px;
    display: flex;
    float: right;
    font-family: 'AnakinMono', sans-serif;
    font-size: 13px;
    height: 50px;
    justify-content: center;
    margin-bottom: 15px;
    width: 130px;

    &:hover {
        background-color: black;
        border-radius: 18px;
        color: #feda4a;
        font-family: 'AnakinMono', sans-serif;
        font-size: 16px;
        height: 60px;
        width: 150px;
        .faveCounter {
            color: white;
        }
    }
}

.faveCounter {
    color: black;
    font-weight: bold;
    margin-left: 5px;
}