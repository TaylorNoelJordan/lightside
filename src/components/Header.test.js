import React from 'react';
import Header from '../components/Header';
import { shallow } from 'enzyme';

describe('Header', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Header />)
    })

    it('should render the movie title and faves button', () => {
        expect(wrapper).toMatchSnapshot();
    })
})