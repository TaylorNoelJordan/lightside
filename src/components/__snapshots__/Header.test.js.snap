// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the movie title and faves button 1`] = `
<header>
  <Link
    to="/"
  >
    <h1
      onClick={[Function]}
    >
      LightSide
    </h1>
  </Link>
  <Link
    to="/Favorites"
  >
    <button
      className="faveBtn"
      onClick={[Function]}
    >
      View Favorites: 
    </button>
  </Link>
</header>
`;
